### PR TITLE
Update Diplomat and remove Clap 3 from lockfile

### PR DESCRIPTION
**Pull request by sffc**
This should resolve some of the security alerts.

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
@robertbastian help, dart CI fails

File: ffi/capi/bindings/dart/FixedDecimal.g.dart [hunk 1]
```diff
@@ -37,7 +37,7 @@ final class FixedDecimal implements ffi.Finalizable {
   /// See the [Rust documentation for `FloatPrecision`](https://docs.rs/fixed_decimal/latest/fixed_decimal/enum.FloatPrecision.html) for more information.
   ///
   /// Throws [Error] on failure.
-  factory FixedDecimal.fromDoubleWithLowerMagnitude(double f, int magnitude) {
+  static FixedDecimal create_from_double_with_lower_magnitude(double f, int magnitude) {
```
Comment (robertbastian) [inline ffi/capi/bindings/dart/FixedDecimal.g.dart:40]: this is probably caused by Diplomat introducing the constructor attribute, so you need to annotate `icu_capi`

Comment (Manishearth) [reply ffi/capi/bindings/dart/FixedDecimal.g.dart:40]: Yes, we should go through and tag stringifiers, constructors, named constructors, and accessors

Comment (sffc) [reply ffi/capi/bindings/dart/FixedDecimal.g.dart:40]: ok so in other words I can't do a clean update of diplomat?

**Comment by sffc** [comment]
I'll close this PR since it didn't do anything other than bump the Diplomat version numbers, and it is not currently possible to do a clean Diplomat update.

**Pull request closed without merging**